app/screens/wines/WineDetailsScreen.tsx

```diff
@@ -1,74 +1,131 @@
-import { useEffect, useState } from 'react';
-import { StyleSheet } from 'react-native';
-import { Picker, Text, View } from 'react-native-ui-lib';
-import { useDispatch, useSelector } from 'react-redux';
-import StoreCardList from '../../components/stores/StoreCardList';
-import { fetchStoresAsync } from '../../features/stores/storesSlice';
-import Store from '../../models/Store';
 import Wine from '../../models/Wine';
+import { Button, Picker, PickerModes, Text, View } from 'react-native-ui-lib';
+import React from 'react';
+import {
+  RefreshControl,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+} from 'react-native';
+import {
+  WineDetailsScreenRouteProp,
+  WinesScreenNavigationProp,
+} from './WinesStackScreen';
+import { useDispatch, useSelector } from 'react-redux';
 import { AppDispatch, RootState } from '../../store/store';
-import { WineDetailsScreenRouteProp } from './WinesStackScreen';
+import Store from '../../models/Store';
+import { useCallback, useEffect, useState } from 'react';
+import { fetchStoresAsync } from '../../features/stores/storesSlice';
+import { useNavigation } from '@react-navigation/native';
+import RatingCardList from '../../components/ratings/RatingCardList';
+import StoreCardList from '../../components/stores/StoreCardList';
+import {
+  fetchWinesAsync,
+  updateStoresForWineAsync,
+} from '../../features/wines/winesSlice';
 
 const WineDetailsScreen: React.FC<{ route: WineDetailsScreenRouteProp }> = ({
   route,
 }): React.ReactElement => {
-  const wine: Wine = route.params.wine;
   const dispatch: AppDispatch = useDispatch();
   const stores: Store[] = useSelector((state: RootState) =>
     state.stores.status !== 'failed' ? state.stores.data : [],
   );
-  const [selectedStores, setSelectedStores] = useState<Store[]>(wine.stores);
+  const wine: Wine = useSelector((state: RootState) =>
```

Comment:
I don’t like this here. I feel like the component should only use the data it's given and not fetch it's own

Reply:
No I’m not sure. I see why the component should for example fetch the stores by itself. I think instead of passing the wine object down, only pass down the id to make clear that the component fetches the wine itself